pkg/k8s/types.go

```diff
@@ -102,14 +109,20 @@ type Controller interface {
 	// GetEndpoints returns the endpoints for a given service, if found
 	GetEndpoints(service.MeshService) (*corev1.Endpoints, error)
 
-	// UpdateStatus updates the status subresource for the given resource and GroupVersionKind
-	// The object within the 'interface{}' must be a pointer to the underlying resource
-	UpdateStatus(interface{}) (metav1.Object, error)
-
 	// GetPodForProxy returns the pod for the given proxy
 	GetPodForProxy(*envoy.Proxy) (*v1.Pod, error)
 
-	GetTargetPortForServicePort(types.NamespacedName, uint16) (uint16, error)
-
 	ServiceToMeshServices(svc corev1.Service) []service.MeshService
 }
+
+// PassthroughInterface is the interface for methods that are implemented by the k8s.Client, but are not considered
+// k8s "specific", and thus do not need further abstraction, and can be used throughout the code base without fear of
+// coupling to k8s.
+type PassthroughInterface interface {
```

Comment:
Consider renaming it to OSMInterface? Since these features are closely tied to OSM.

Reply:
the `compute.Interface` is essentially that (which contains the PassthroughInterface), The passthrough is a subset of what would be the OSMInterface, in that it is what we allow to leak from k8s.

Not sure what the best answer is so I'll leave this comment unresolved to see if others have opinions on naming here :)

Reply:
Ok, I think I initially misunderstood compute.Interface which I thought it should contain methods that are cluster provider related and should have nothing to do with OSM terminologies.